src/components/PostEditor/PostEditor.connector.js

```diff
@@ -64,15 +64,13 @@ export function mapStateToProps (state, props) {
   const topic = getTopicForCurrentRoute(state, props)
   const topicName = get('name', topic)
   const announcementSelected = state[MODULE_NAME].announcement
-  const canModerate = currentUser && currentUser.canModerate(currentGroup)
```

Comment:
hmm, why is this not working? we use this function all over the app, why is it not working here? seems better to use common shared code than do something unique here

Reply:
The use case it didn't work for here is that the user needs to be able to moderate all groups associated with a post, not just the original one that was selected when the PostEditor was opened.

Since the state of the PostEditor can change, the check for this was pushed down to that level, where it has access to the groups selected for the post. The post can only be an announcement if ALL groups selected are moderated by the user.

Reply:
sooo, what should actually happen here? I actually think you should be able to choose announcement still, but the announcement will only get sent to groups the user moderates. @lorenjohnson what do you think? in which case this issue gets pushed back to the back-end. Though we still need to check whether any groups can be moderated to decide whether to show the announcement button, so this change is still needed

Reply:
Yeah, let me know what the group consensus is here. @lorenjohnson if you have a view?

Reply:
I would either only allow announcement in the case of a single group selection in which the user is a moderator, or allow it if the user is a moderator of all the selected groups. Either one I'm fine with.

Showing the announcement option for a mixed set of groups currently I think might be a little confusing or misleading, without a tiny bit more thought/work on the UI feedback around the feature. I don't think it's worth that at this moment and can be revisited if we get feedback or need from group moderators or otherwise we are really feeling it.

Reply:
It looks like "allow if the user is a moderator of all the selected groups" is what you've done here @thomasgwatson, so I'm going to go ahead give give my own PR approval :)